### DISCUSSION TITLE
Create image from raw pixels

### DISCUSSION THREAD

**Discussion by nearwall** [jmjoy/graphicsmagick-rs#20] [Q&A] [answered]
Hello!
Is it possible to create an image from raw pixels like in Magick++ [documentation](http://www.graphicsmagick.org/Magick++/Image.html#id11)??:
```
Image ( const unsigned int width_,
        const unsigned int height_,
        const std::string &map_,
        const StorageType type_,
        const void *pixels_ )

 ```
 I was trying do it with [MagicWand](https://docs.rs/graphicsmagick/latest/graphicsmagick/wand/magick/struct.MagickWand.html#).
But [read_image_blob](https://docs.rs/graphicsmagick/latest/graphicsmagick/wand/magick/struct.MagickWand.html#method.read_image_blob) doesn't work with pixels and fails, and I don't see any opportunity to specify `MagicWand` to work with raw pixels.

Thanks in advance for any suggestions !

**Comment by NobodyXu** [comment, marked as answer]
It's definitely possible, I'm planning to submit a PR for it.

**Reply by NobodyXu** [reply]
It should be within one day or two.

**Reply by nearwall** [reply]
@NobodyXu thanks a lot for your answers!

This works for me:
``` 
       # create empty Wand
        let mut wand = MagickWand::new();
        # set format(for example RGB)
        wand.set_format()
        # set image size
        wand.set_size()
        # read raw pixels
        wand.read_image_blob()
```
